### ISSUE TITLE
op/lc.t

### ISSUE THREAD

**Issue by toddr** [CpanelInc/perl-compiler#47]
This is the minimum I've been able to reduce this one.

```
BEGIN {
    require './test.pl';
}

plan(tests => 63);

# The return statement should make no difference in this case:
sub not_constant () {        42 }
use feature 'lexical_subs'; no warnings 'experimental::lexical_subs';
${\not_constant}++;

# We do not want re.pm loaded at this point.  Move this test up or find
# another XSUB if this fails.
{
    sub re::regmust{}
    bless \&re::regmust;
    DESTROY {
        no warnings 'redefine', 'prototype';
        my $str1 = "$_[0]";
        *re::regmust = sub{}; # GvSV had no refcount, so this freed it
        my $str2 = "$_[0]";   # used to be UNKNOWN(0x7fdda29310e0)
        @str = ($str1, $str2);
    }
    local $^W; # Suppress redef warnings in XSLoader
    require re;
    is $str[1], $str[0],
      'XSUB clobbering sub whose DESTROY assigns to the glob';
}
```

```
#0  0x00007ffff787cfe4 in Perl_refcounted_he_chain_2hv (chain=0x748840, flags=0) at hv.c:3163
#1  0x00007ffff78dd2c9 in Perl_pp_caller () at pp_ctl.c:1951
#2  0x00007ffff788059b in Perl_runops_standard () at run.c:41
#3  0x00007ffff77ba477 in S_run_body (oldscope=1) at perl.c:2485
#4  0x00007ffff77b9f54 in perl_run (my_perl=0x6f2118 <BC_unthreaded_interpreter>) at perl.c:2408
#5  0x000000000041902d in main (argc=1, argv=0x7fffffffe7f8, env=0x7fffffffe808) at /usr/local/cpanel/B-C/t/v5.24.1/t/op/sub.c:38711
```

```
  >│3163            U32 hash = HEK_HASH(chain->refcounted_he_hek);            
```

```
(gdb) p	(chain->refcounted_he_hek)
$2 = (HEK *) 0x31
```

**Comment by toddr** [comment]
```
use feature 'lexical_subs'; 
no warnings 'experimental::lexical_subs';
my sub not_constantm () { 42 }
sub do_caller { my @caller = caller(1) }
sub x { do_caller() }
{
    sub re::regmust{}
    bless \&re::regmust; # force DESTROY
    #DESTROY { print qq{DESTROYED\n} }
    no warnings;
    require re;
    x();
}
print qq[ok\n];

**Comment by toddr** [comment]
Fixed by f4c2400b9cf0b92020e5e5374af6dca0d0e563c0
